### PR TITLE
Update migration notes

### LINKED ISSUE #50: Adjust migration notes after merge
We have to describe the merge and how to migrate from OSIAM 2.x in the respective documents.

### PR DESCRIPTION
**Pull request by tkrille**
Resolves #50

### CODE REVIEW

File: docs/migration.md [hunk 1]
```diff
@@ -1,24 +1,258 @@
...
 
-## from 2.2 to 3.0
+## From 2.5 to 3.0
 
-This release removes the old, HTTP method-based scopes and replaces them with
-the new, functional-motivated scopes `ADMIN` and `ME`. OSIAM will
-automatically migrate all well-known clients to the new scopes in the following
-way:
+In this release, the auth-server and resource-server have been merged into a single application.
+Therefore you need to merge the databases and configuration files, too.
```
Comment (wallner) [inline docs/migration.md]: Is the comma really correct?

File: docs/migration.md [hunk 2]
```diff
@@ -1,24 +1,258 @@
...
 
-Well-known clients and their scopes are:
+- You are using the latest version of OSIAM 2.x
+- You are using the latest version of the add-ons (if applicable)
+- Your Java Applications are using at least version 1.8 of the Connector4Java.
 
-* `example-client` (`ADMIN`, `ME`)
-* `addon-self-administration-client` (`ADMIN`)
-* `addon-administration-client` (`ADMIN`)
+**NOTE:** It's highly recommended to try the migration on a staging system, before updating a production system.
```
Comment (wallner) [inline docs/migration.md]: Please write out 'It is'

**Comment by tkrille** [comment]
Thanks for your comments @wallner. I've updated the PR.

File: docs/migration.md [hunk 3]
```diff
@@ -1,24 +1,258 @@
...
+Think about the ways of deployment for a minute and read the [documentation]
+(detailed-reference-installation.md#starting-osiam) on this.
+If you decide to run OSIAM as a standalone application from now on, follow the instructions in the documentation.
+
+**Note:** The distribution artifacts have been dropped.
+Everything you need to install OSIAM is contained in `osiam.war` now.
+
+### Merge the Databases
+
+If you are using 2 separated databases, you will have to merge them now into 1 database.
```
Comment (wallner) [inline docs/migration.md]: "If you are using two separate databases"

File: docs/migration.md [hunk 4]
```diff
@@ -1,24 +1,258 @@
...
+You can actually do this without updating OSIAM afterwards, as OSIAM 2.x already supports a single database.
```
Comment (wallner) [inline docs/migration.md]: I think: "as OSIAM 2.x already supported a single database" is better.

File: docs/migration.md (hunk 3 above)
Comment (tkrille) [inline docs/migration.md]: And also for the "into 1 database"?

**Comment by tkrille** [comment]
Updated

File: docs/migration.md [hunk 5]
```diff
@@ -1,24 +1,258 @@
...
+### Choose a Way of Deployment
+
+You probably deployed OSIAM in a Tomcat Servlet container before this update.
+This is still supported, but now there also is the option to run OSIAM as a standalone application.
+Think about the ways of deployment for a minute and read the [documentation]
+(detailed-reference-installation.md#starting-osiam) on this.
+If you decide to run OSIAM as a standalone application from now on, follow the instructions in the documentation.
+
+**Note:** The distribution artifacts have been dropped.
+Everything you need to install OSIAM is contained in `osiam.war` now.
```
Comment (wallner) [inline docs/migration.md]: "Everything you need to install OSIAM is now contained in `osiam.war`.

File: docs/migration.md [hunk 6]
```diff
@@ -1,24 +1,258 @@
...
+
+### Merge the Databases
+
+If you are using two separated databases, you will have to merge them now into one database.
+You can actually do this without updating OSIAM afterwards, as OSIAM 2.x already supported a single database.
+Verify, that everything is still working, after the merge.
```
Comment (wallner) [inline docs/migration.md]: both commas are wrong, I think.

File: docs/migration.md [hunk 7]
```diff
@@ -1,24 +1,258 @@
...
+
+```sh
+# For standalone deployments
+curl http://localhost:8080/ServiceProviderConfig
+
+# For Tomcat deployments
+curl http://localhost:8080/osiam/ServiceProviderConfig
+```
+
+If OSIAM is running correctly, you'll get the following (or similar) response:
```
Comment (wallner) [inline docs/migration.md]: "you will get the following (or a similar) response:"

File: docs/migration.md [hunk 8]
```diff
@@ -1,24 +1,258 @@
...
+
+# Tomcat deployment
+org.osiam.home=http://localhost:8080/osiam
+```
+
+### Update your Applications
+
+#### Update Scopes
+
+This release removes support the deprecated HTTP-method-based scopes.
```
Comment (wallner) [inline docs/migration.md]: "This release removes support of the deprecated HTTP method based scopes."

File: docs/migration.md [hunk 9]
```diff
@@ -1,24 +1,258 @@
...
+They are replaced with the new, functional-motivated scopes `ADMIN` and `ME`.
```
Comment (wallner) [inline docs/migration.md]: "functionally motivated scopes"?

File: docs/migration.md [hunk 10]
```diff
@@ -26,13 +260,52 @@ your existing clients:
...
-5. Deploy OSIAM
 
-**NOTE:** The new scopes `ADMIN` and `ME` have been supported since the last
-release, to make it possible to just switch to them without upgrading OSIAM and
-still have everything work as expected. This way you can schedule an upgrade at
-your own convenience and minimize downtime.
+**WARNING:** Adding the scope `ADMIN` to an untrusted client can lead to security issues.
+Access tokens granted to it will have full access to OSIAM.
+
+**NOTE:** Remember to change your applications, too, so they use the new scopes.
```
Comment (wallner) [inline docs/migration.md]: drop the ", too,"

File: docs/migration.md [hunk 11]
```diff
@@ -26,13 +260,52 @@ your existing clients:
...
+
+The new scopes `ADMIN` and `ME` are supported since version 2.1.
```
Comment (wallner) [inline docs/migration.md]: "have been supported since"

File: docs/migration.md [hunk 12]
```diff
@@ -26,13 +260,52 @@ your existing clients:
...
+This way it is possible to switch without updating OSIAM and still have everything work as expected.
+This way you can schedule an update at your own convenience and minimize downtime.
+
+#### Configure the Connector4Java
+
+If you write Java applications that integrate with OSIAM and are using the Connector4Java,
+then you will have to change the configuration of the connector:
```
Comment (wallner) [inline docs/migration.md]: "then you need to change the configuration"

File: docs/migration.md [hunk 13]
```diff
@@ -26,13 +260,52 @@ your existing clients:
...
+        .withEndpoint(OSIAM_ENDPOINT)
+        [...]
+        .build();
+```
+
+#### API Changes
+
+There have been some minor API changes, namely:
+
+- Reply with 400 BAD REQUEST to invalid filters.
```
Comment (wallner) [inline docs/migration.md]: Invalid filters now get a reply of 400 BAD REQUEST.

File: docs/migration.md [hunk 14]
```diff
@@ -26,13 +260,52 @@ your existing clients:
...
+- Change URL of service provider configuration resource from `/ServiceProviderConfigs` to `/ServiceProviderConfig`.
```
Comment (wallner) [inline docs/migration.md]: The service provider configuration was changed from ... to ...

File: docs/migration.md [hunk 15]
```diff
@@ -26,13 +260,52 @@ your existing clients:
...
+
+Update your applications accordingly.
```
Comment (wallner) [inline docs/migration.md]: Please update your applications accordingly.

File: docs/migration.md (hunk 9 above)
Comment (tkrille) [inline docs/migration.md]: I think functionally-motivated is the right one.

File: docs/migration.md (hunk 13 above)
Comment (tkrille) [inline docs/migration.md]: Thanks for the rewording. I actually just copied the entry from the changelog :)

**Comment by tkrille** [comment]
Thanks again @wallner. I've updated the PR.

**Pull request merged**